### PR TITLE
Split config into separate package

### PR DESCRIPTION
**Pull request by jacobwinch**
This PR moves code related to config into a separate package.

### CODE REVIEW

File: lib/handler/src/main/scala/com/gu/util/config/LoadConfig.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    s3Load: Try[String]
+  ): FailableOp[Config[StepsConfig]] = {
+    logger.info(s"${this.getClass} Lambda is starting up in $stage")
+    for {
+      textConfig <- s3Load.toFailableOp("load config from s3")
+      config <- parseConfig(textConfig).toFailableOp("parse config file")
+      _ <- if (stage == config.stage)
+        \/-(())
+      else
+        -\/(ApiGatewayResponse.internalServerError(s"running in $stage with config from ${config.stage}"))
```
Comment (jacobwinch) [inline lib/handler/src/main/scala/com/gu/util/config/LoadConfig.scala]: We still need another small PR to make this return type non-API Gateway specific.

File: lib/handler/src/main/scala/com/gu/util/config/LoadConfig.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package com.gu.util.config
```
Comment (johnduffell) [inline lib/handler/src/main/scala/com/gu/util/config/LoadConfig.scala]: the api gateway stuff shouldn't be in the jar if it's not needed - perhaps rename the lib/handler to lib/apigateway and move this to a new lib/config
Then the (lack of) interdependencies will be enforced by the compiler, and the structure will be clearer.
I'm interested in using something like https://www.pantsbuild.org/ to make the process of creating separate source folders easier, or learning to use sbt better.

Comment (johnduffell) [reply lib/handler/src/main/scala/com/gu/util/config/LoadConfig.scala]: of course that could all be in a second PR which depends on a fix for the below comment about the API gateway speciific type!

File: handlers/digital-subscription-expiry/src/test/scala/com/gu/digitalSubscriptionExpiry/zuora/GetAccountSummaryEffectsTest.scala [hunk 3]
```diff
@@ -16,8 +16,8 @@ class GetAccountSummaryEffectsTest extends FlatSpec with Matchers {
     val testAccountId = AccountId("2c92c0f86078c4d4016079e1402d6536")
 
     val actual: \/[io.Serializable, AccountSummaryResult] = for {
-      configAttempt <- ConfigLoad.load(Stage("DEV")).toEither.disjunction
-      config <- Config.parseConfig[StepsConfig](configAttempt)
+      configAttempt <- S3ConfigLoad.load(Stage("DEV")).toEither.disjunction
+      config <- LoadConfig.parseConfig[StepsConfig](configAttempt)
```
Comment (johnduffell) [inline handlers/digital-subscription-expiry/src/test/scala/com/gu/digitalSubscriptionExpiry/zuora/GetAccountSummaryEffectsTest.scala]: good name improvements

**Comment by johnduffell** [comment]
great, good work! 👍 🚀 
It's a shame we have pretty much the same 2 lines in all the effects tests, I'm sure we could tidy that up, but that's nothing new.

**Pull request merged**